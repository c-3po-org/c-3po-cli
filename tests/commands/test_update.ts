import * as path from "path";
import * as fs from "fs";
import { execSync } from "child_process";
import * as tmp from "tmp";

const originalPo = `msgid ""
msgstr ""
"Content-Type: text/plain; charset=utf-8\n"
"Plural-Forms: nplurals=2; plural=(n!=1);\n"


msgid "old"
msgstr "old trans"

msgid "obsolete"
msgstr "obsolete trans"
`;

const srcPath = path.resolve(__dirname, "../fixtures/updateTest/test.js");

test("test update po", () => {
    const tmpFile = tmp.fileSync();
    fs.writeFileSync(tmpFile.name, originalPo);
    execSync(`ts-node src/index.ts update ${tmpFile.name} ${srcPath}`);
    const result = fs.readFileSync(tmpFile.name).toString();
    expect(result).toMatchSnapshot();
    tmpFile.removeCallback();
});

test("test for plugin override", () => {
    const tmpFile = tmp.fileSync();
    fs.writeFileSync(tmpFile.name, originalPo);
    execSync(
        `ts-node src/index.ts update --discover=_  ${tmpFile.name} ${srcPath}`
    );
    const result = fs.readFileSync(tmpFile.name).toString();
    expect(result).toContain('msgid "discover _ test"');
    tmpFile.removeCallback();
});

test("test update with multiple discover po (plugins settings override test)", () => {
    const tmpFile = tmp.fileSync();
    fs.writeFileSync(tmpFile.name, originalPo);
    execSync(
        `ts-node src/index.ts update --discover=_ --discover=gettext ${
            tmpFile.name
        } ${srcPath}`
    );
    const result = fs.readFileSync(tmpFile.name).toString();
    expect(result).toContain('msgid "discover _ test"');
    tmpFile.removeCallback();
});

const commentsTest = path.resolve(
    __dirname,
    "../fixtures/updateTest/comments.jsx"
);
test("should extract comments by default", () => {
    const tmpFile = tmp.fileSync();
    fs.writeFileSync(tmpFile.name, originalPo);
    execSync(`ts-node src/index.ts update ${tmpFile.name} ${commentsTest}`);
    const result = fs.readFileSync(tmpFile.name).toString();
    expect(result).toContain("#. translator: test comment");
    expect(result).toContain("#. translator: jsx test comment");
    tmpFile.removeCallback();
});
