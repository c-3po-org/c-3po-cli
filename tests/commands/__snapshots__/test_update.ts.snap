// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should sort the output alphabetically (apply sortByMsgid option) 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"

#: tests/fixtures/updateTest/test.js:6
#, javascript-format
msgid \\"\${ n } banana\\"
msgid_plural \\"\${ n } bananas\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\"

#: tests/fixtures/updateTest/test.js:4
msgid \\"new\\"
msgstr \\"\\"

#: tests/fixtures/updateTest/test.js:3
msgid \\"test\\"
msgstr \\"\\""
`;

exports[`test update po 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"

#: tests/fixtures/updateTest/test.js:3
msgid \\"test\\"
msgstr \\"\\"

#: tests/fixtures/updateTest/test.js:4
msgid \\"new\\"
msgstr \\"\\"

#: tests/fixtures/updateTest/test.js:6
#, javascript-format
msgid \\"\${ n } banana\\"
msgid_plural \\"\${ n } bananas\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;
